src/main/java/io/github/moulberry/notenoughupdates/profileviewer/BasicPage.java

```diff
@@ -565,8 +565,39 @@ public String getSkinType() {
 		}
 
 		GlStateManager.color(1, 1, 1, 1);
-		JsonObject petsInfo = profile.getProfile(profileName).getPetsInfo();
+		SkyblockProfiles.SkyblockProfile currentProfile = profile.getProfile(profileName);
+		JsonObject petsInfo = currentProfile.getPetsInfo();
 		if (petsInfo != null) {
+			if (currentProfile.getGamemode() != null && currentProfile.getGamemode().equals("bingo")) {
+				JsonArray pets = petsInfo.get("pets").getAsJsonArray();
+				for (JsonElement pet : pets) {
+					if (pet.getAsJsonObject().get("type").getAsString().equals("BINGO")) {
+						String tier = pet.getAsJsonObject().get("tier").getAsString();
```

Comment:
I think there could be a few more null checks here.

Reply:
Every bingo profile should have a bingo pet but I can add some more tomorrow